presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ReorderJoins.java

```diff
@@ -146,17 +150,30 @@ public int compare(PlanNode node1, PlanNode node2)
         }
 
         private PlanNode chooseJoinOrder(List<PlanNode> sources, List<Symbol> outputSymbols)
+                throws UnknownCostException
```

Comment:
I decided to make JoinEnumerator only choose plans among plans  that have known costs.  otherwise it throws an UnknownCostException.  building error handling for this case into the JoinEnumerator code was very complicated.  Particularly because Optional.empty() is already used throughout for cross joins.  As a result, we would actually need to be able to return one of three things from all the recursive creatJoin() calls.
a) a plan 
b) something indicate that computing a join would require a cross join. If there are no valid join options at the end, we then want to introduce a cross join
c) something to indicate that we encountered a plan who's cost we can't evaluate.  From here we want to bail out of the whole enumeration and return Optional.empty()  (we would need to change the return type of chooseJoinOrder to be an optional). 

 Creating some kind of object to handle that would be pretty gross, and also seems unnecessarily complicated since basically what we mean by c, is that these types of plans aren't supported.